### PR TITLE
EVG-16490: Don't return 500s when getting test results fails.

### PR DESCRIPTION
**Pull request by johndaniels**
Currently we don't return the error, but we also send an error
response globally, causing this to 500 in the event of a Cedar
outage. This changes the logic so that we will return no test
results if Cedar is down but otherwise continue to process the
response.

[EVG-16490](https://jira.mongodb.org/browse/EVG-16490)

### Description 
This updates getTestResults to not send a 500 response if for some reason Cedar is down. It now silently logs the error and the request continues to process normally.

### Testing 
  I added testing to make sure that we do in fact return a nil list of results if Cedar is unavailable. Also, the function no longer has access to the response and request objects so it is impossible for it to send a 500 response code header.

### CODE REVIEW

File: service/task.go [hunk 1]
```diff
@@ -911,7 +914,11 @@ func (uis *UIServer) getTestResults(w http.ResponseWriter, r *http.Request, proj
 				et, err = task.FindOneId(t)
 			}
 			if err != nil {
-				uis.LoggedError(w, r, http.StatusInternalServerError, err)
+				grip.Error(message.Fields{
+					"message": "fetching execution task",
```
Comment (julianedwards) [inline service/task.go:918]: `fetching test results for execution task`

File: service/task.go [hunk 2]
```diff
@@ -911,7 +914,11 @@ func (uis *UIServer) getTestResults(w http.ResponseWriter, r *http.Request, proj
...
+					"task":    t,
```
Comment (julianedwards) [inline service/task.go:919]: `task_id`

File: service/task.go [hunk 3]
```diff
@@ -911,7 +914,11 @@ func (uis *UIServer) getTestResults(w http.ResponseWriter, r *http.Request, proj
...
+					"parent":  projCtx.Task.Id,
```
Comment (julianedwards) [inline service/task.go:920]: `parent_task_id`

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: service/task_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	uis := UIServer{hostCache: make(map[string]hostCacheItem)}
+	projectContext := projectContext{
+		Context: model.Context{
+			Task: &newTask,
+		},
+	}
+	uiTask := uiTaskData{}
+	results := uis.getTestResults(projectContext, &uiTask)
+	var expected []task.TestResult = nil
+	assert.Equal(t, expected, results)
```
Comment (ablack12) [inline service/task_test.go:37]: assert.Nil

File: service/task.go [hunk 5]
```diff
@@ -891,9 +891,12 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
...
 
-func (uis *UIServer) getTestResults(w http.ResponseWriter, r *http.Request, projCtx projectContext, uiTask *uiTaskData) []task.TestResult {
+func (uis *UIServer) getTestResults(projCtx projectContext, uiTask *uiTaskData) []task.TestResult {
 	if err := projCtx.Task.PopulateTestResults(); err != nil {
-		uis.LoggedError(w, r, http.StatusInternalServerError, err)
+		grip.Error(message.WrapError(err, message.Fields{
+			"task_id": projCtx.Task.Id,
+			"message": "fetching test results for task",
+		}))
 		return nil
```
Comment (ablack12) [inline service/task.go:900]: Should we initialize uis.TestResults above? (Is there concern around nil pointer errors if we aren't sending an error? Didn't read back where this is used though.

**Review by ablack12 — APPROVED** [review]

File: service/task.go (hunk 5 above)
Comment (johndaniels) [reply service/task.go:900]: The only place it gets used in Go code is as part of append inside that function. After that I _think_ it goes to JS code, which could have an issue with null maybe? 

I don't see any JS errors in the console but all I'm doing is loading the task page and that seems to work.

Comment (johndaniels) [reply service/task.go:900]: Happy to change it though if it seems like a good idea.

File: service/task.go (hunk 1 above)
Comment (johndaniels) [reply service/task.go]: fixed

File: service/task.go (hunk 2 above)
Comment (johndaniels) [reply service/task.go]: fixed

File: service/task.go (hunk 3 above)
Comment (johndaniels) [reply service/task.go]: fixed

**Review by julianedwards — APPROVED** [review]

**Pull request merged**